src/anm/render.js

```diff
@@ -63,6 +68,13 @@ function r_loop(ctx, player, anim, before, after, before_render, after_render) {
         if (!after(time)) return;
     }
 
+    //increase the counter and fire the event if necessary
+    timeCounters[player.id] += dt;
+    if (timeCounters[player.id] >= TIME_UPDATE_TIME) {
```

Comment:
indeed, sounds better, my first implementation used a number of frames to fire the event, hence the strange name